### PR TITLE
[EFR32] Window Covering app modified to compile on boards without LCD.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
The Window Covering app is not compiling for BRD4166a.

#### Change overview
Guards added to remove the LCD-dependent code from the app.

#### Testing
Tested on BRD4164a (WSTK) and BRD4166a (TBS2).

### CODE REVIEW

**Review by jepenven-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
master 1.0 approval reason: fixes compilation of existing example.

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 1]
```diff
@@ -129,16 +132,17 @@ StaticQueue_t sAppEventQueueStruct;
...
 {
     return WindowAppImpl::sInstance;
 }
 
-WindowAppImpl::WindowAppImpl() : mIconTimer("Timer:icon", LCD_ICON_TIMEOUT, OnIconTimeout, this) {}
+WindowAppImpl::WindowAppImpl()
+#ifdef DISPLAY_ENABLED
+    :
+    mIconTimer("Timer:icon", LCD_ICON_TIMEOUT, OnIconTimeout, this)
+#endif
```
Comment (jmeg-sfy) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:144]: this is itchy to my eyes..

File: examples/window-app/efr32/src/WindowAppImpl.cpp [hunk 2]
```diff
@@ -129,16 +132,17 @@ StaticQueue_t sAppEventQueueStruct;
...
+{}
```
Comment (jmeg-sfy) [inline examples/window-app/efr32/src/WindowAppImpl.cpp:145]:
```suggestion
#ifdef DISPLAY_ENABLED
WindowAppImpl::WindowAppImpl() : mIconTimer("Timer:icon", LCD_ICON_TIMEOUT, OnIconTimeout, this) {}
#else
WindowAppImpl::WindowAppImpl()
#endif
```

**Review by jmeg-sfy — APPROVED** [review]

File: examples/window-app/efr32/src/WindowAppImpl.cpp (hunk 2 above)
Comment (rcasallas-silabs) [reply examples/window-app/efr32/src/WindowAppImpl.cpp]: Certainly better. Suggestion applied.

**Pull request merged**